test/_keys/Keys.php

```diff
@@ -21,25 +20,29 @@ trait Keys
     public static function createRsaKeys(): void
     {
         static::$rsaKeys = [
-            'private'           => LocalFileReference::file(__DIR__ . '/rsa/private.key'),
-            'public'            => LocalFileReference::file(__DIR__ . '/rsa/public.key'),
-            'encrypted-private' => LocalFileReference::file(__DIR__ . '/rsa/encrypted-private.key', 'testing'),
-            'encrypted-public'  => LocalFileReference::file(__DIR__ . '/rsa/encrypted-public.key'),
+            'private'           => Key\InMemory::file(__DIR__ . '/rsa/private.key'),
+            'public'            => Key\InMemory::file(__DIR__ . '/rsa/public.key'),
+            'encrypted-private' => Key\InMemory::file(__DIR__ . '/rsa/encrypted-private.key', 'testing'),
+            'encrypted-public'  => Key\InMemory::file(__DIR__ . '/rsa/encrypted-public.key'),
+            'private_short'     => Key\InMemory::file(__DIR__ . '/rsa/private_512.key'),
+            'public_short'      => Key\InMemory::file(__DIR__ . '/rsa/public_512.key'),
         ];
     }
 
     /** @beforeClass */
     public static function createEcdsaKeys(): void
     {
         static::$ecdsaKeys = [
-            'private'        => LocalFileReference::file(__DIR__ . '/ecdsa/private.key'),
-            'private-params' => LocalFileReference::file(__DIR__ . '/ecdsa/private2.key'),
-            'public1'        => LocalFileReference::file(__DIR__ . '/ecdsa/public1.key'),
-            'public2'        => LocalFileReference::file(__DIR__ . '/ecdsa/public2.key'),
-            'public-params'  => LocalFileReference::file(__DIR__ . '/ecdsa/public3.key'),
-            'private_ec512'  => LocalFileReference::file(__DIR__ . '/ecdsa/private_ec512.key'),
-            'public_ec512'   => LocalFileReference::file(__DIR__ . '/ecdsa/public_ec512.key'),
-            'public2_ec512'  => LocalFileReference::file(__DIR__ . '/ecdsa/public2_ec512.key'),
+            'private'        => Key\InMemory::file(__DIR__ . '/ecdsa/private.key'),
+            'private-params' => Key\InMemory::file(__DIR__ . '/ecdsa/private2.key'),
+            'public1'        => Key\InMemory::file(__DIR__ . '/ecdsa/public1.key'),
+            'public2'        => Key\InMemory::file(__DIR__ . '/ecdsa/public2.key'),
+            'public-params'  => Key\InMemory::file(__DIR__ . '/ecdsa/public3.key'),
+            'private_ec512'  => Key\InMemory::file(__DIR__ . '/ecdsa/private_ec512.key'),
+            'public_ec512'   => Key\InMemory::file(__DIR__ . '/ecdsa/public_ec512.key'),
+            'public2_ec512'  => Key\InMemory::file(__DIR__ . '/ecdsa/public2_ec512.key'),
+            'private_short'  => Key\InMemory::file(__DIR__ . '/ecdsa/private_ec160.key'),
```

Comment:
This fails to load on my machine... OpenSSL says it's unsupported :thinking: 
I'm running `OpenSSL 3.0.5 5 Jul 2022 (Library: OpenSSL 3.0.5 5 Jul 2022)`:

```
openssl
OpenSSL support => enabled
OpenSSL Library Version => OpenSSL 3.0.5 5 Jul 2022
OpenSSL Header Version => OpenSSL 3.0.3 3 May 2022
Openssl default config => /etc/pki/tls/openssl.cnf
openssl.cafile => no value => no value
openssl.capath => no value => no value
Native OpenSSL support => enabled
```

Reply:
```sh
$ openssl ecparam -list_curves
  secp224r1 : NIST/SECG curve over a 224 bit prime field
  secp256k1 : SECG curve over a 256 bit prime field
  secp384r1 : NIST/SECG curve over a 384 bit prime field
  secp521r1 : NIST/SECG curve over a 521 bit prime field
  prime256v1: X9.62/SECG curve over a 256 bit prime field
```

That explains :smile:

Reply:
I guess both `shivanmatur/php` and me run OpenSSL v1.

Can you try `secp192k1` then?

```shell
openssl ecparam -name secp192k1 -genkey -noout -out private_ec192.key
```

Reply:
> That explains

Indeed I would have loved to check key bits before loading it with OpenSSL, but that's out of the scope of this library.

Reply:
OpenSSL v3 removed everything < 224 =)
We can skip the tests for OpenSSL >= 3.0